gflownet/utils/buffer.py

```diff
@@ -189,6 +201,12 @@ def make_data_set(self, config):
             and hasattr(self.env, "get_uniform_terminating_states")
         ):
             samples = self.env.get_uniform_terminating_states(config.n, config.seed)
+        elif (
+            config.type == "random"
+            and "n" in config
+            and hasattr(self.env, "get_random_terminating_states")
```

Comment:
I wouldn't bother with changing this in this PR, but a comment for the evaluation refactor: I would add a better exception handling, i.e. only check if `config.type == ...`, and do the remaining checks within that if clause (and raise appropriate errors if, for example, `n` is not in config). The reason being, from a user perspective it might not always be clear why we are not getting any evaluation, while we are clearly requesting one.

Reply:
you are right, adding an issue